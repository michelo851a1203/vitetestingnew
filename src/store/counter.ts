import { defineStore } from "pinia";

export interface UseCounterStoreType {
  currentCount: number;
}

export const useCounterStore = defineStore('counter', {
  state: () => {
    return <UseCounterStoreType>{
      currentCount: 0,
    };
  },
  actions: {
    appendCount() {
      this.currentCount++;
    },
  }
})

