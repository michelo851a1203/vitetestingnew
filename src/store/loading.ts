import { defineStore } from "pinia";
import { LoadingStatus } from "../datatype/loading";

export interface UseLoadingStoreType {
  isLoadingArray: LoadingStatus[],
}

export const useLoadingStore = defineStore("loading", {
  state: () => {
    return <UseLoadingStoreType>{
      isLoadingArray: [],
    }
  },
  actions: {
    isNowLoading(loadingStatus: LoadingStatus) {
      return this.isLoadingArray.includes(loadingStatus);
    },
    addLoadingStatus(loadingStatus: LoadingStatus) {
      this.isLoadingArray.push(loadingStatus);
    },
    removeLoadingStatus(loadingStatus: LoadingStatus) {
      const loadingIndex = this.isLoadingArray.findIndex((loadingStatusItem) => {
        return loadingStatusItem === loadingStatus;
      });
      this.isLoadingArray.splice(loadingIndex, 1);
    }
  }
})
