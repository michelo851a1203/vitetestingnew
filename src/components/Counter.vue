<script setup lang="ts">
import { storeToRefs } from 'pinia';
import { useCounterStore } from '../store/counter';

  const {
    appendCount,
  } = useCounterStore()

  const {
    currentCount,
  } = storeToRefs(useCounterStore());
  
</script>

<template>
  <button
    @click="appendCount"
    class="px-4 py-3 rounded-xl bg-blue-300 border-none"
  >
    {{ currentCount }}
  </button>
</template>
