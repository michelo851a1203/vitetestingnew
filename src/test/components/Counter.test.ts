import { describe ,test, expect } from 'vitest';
import Counter from '../../components/Counter.vue'
import { mount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';
import { useCounterStore } from '../../store/counter';
import { storeToRefs } from 'pinia';

describe('testing counter component', () => {
  const wrapper = mount(Counter, {
    global: {
      plugins: [createTestingPinia({
        stubActions: false,
      })]
    },
  });

  test('mount counter component initial', async () => {
    expect(wrapper.html()).toMatchSnapshot();
    expect(Counter).toBeTruthy();
    expect(wrapper.text()).toContain(0);
  });

  test('click button', async () => {
    const {
      appendCount,
    } = useCounterStore()

    wrapper.get('button').trigger('click');
    expect(appendCount).toHaveBeenCalledTimes(1);
    await wrapper.vm.$nextTick();
    expect(wrapper.text()).toContain(1);
  });

  test('double click', async () => {
    const {
      appendCount,
    } = useCounterStore()

    wrapper.get('button').trigger('click');
    expect(appendCount).toHaveBeenCalledTimes(2);
    await wrapper.vm.$nextTick();
    expect(wrapper.text()).toContain(2);
  });
});
