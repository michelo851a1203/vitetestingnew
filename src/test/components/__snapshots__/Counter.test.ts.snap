// Vitest Snapshot v1

exports[`testing counter component > mount counter component initial 1`] = `"<button class=\\"px-4 py-3 rounded-xl bg-blue-300 border-none\\">0</button>"`;
